keps/sig-instrumentation/2831-kubelet-tracing/README.md

```diff
@@ -140,6 +141,14 @@ to generate spans for sampled incoming requests and propagate context with clien
 
 OpenTelemetry-Go provides the [propagation package](https://github.com/open-telemetry/opentelemetry-go/blob/main/propagation/propagation.go) with which you can add custom key-value pairs known as [baggage](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/baggage/api.md). Baggage data will be propagated across services within contexts.
 
+### Connected Traces with Nested Spans
+
+Nested spans with top-level traces in the kubelet will connect CRI calls together. Nested spans will be created for the following:
+* Sync Loops (e.g. syncPod, eviction manager, various gc routines) where the kubelet initiates new work.
+    * [top-level traces for pod sync and GC](https://github.com/kubernetes/kubernetes/pull/114504)
+* Incoming requests (exec, attach, port-forward, metrics endpoints, podresources)
+* Outgoing requests (CNI, CSI, device plugin, k8s API calls)
```

Comment:
In general this means we have to go manually through the kubelet code and decide on a case-by-case basis if it makes sense to add another span or not, right?

We can definitely work as a team on that since it's a great opportunity to learn more about the kubelet code.

Reply:
Ideally we would use instrumentation libraries (e.g. grpc, http) to instrument these connections

Reply:
@saschagrunert agreed & definitely plan to work as a team to implement next phase of kubelet tracing, this has been a journey of getting to know the kubelet code & OTel. I'm thankful @dashpole has been guiding this  :)